### ISSUE TITLE
Add event handler attributes (e.g. onsourceopen) to MediaSource, SourceBuffer and SourceBufferList

### ISSUE THREAD

**Issue by foolip** [w3c/media-source#66]
This is useful for feature detection, and is commonly done for things that inherit from `EventTarget`:
https://w3c.github.io/web-animations/#the-animation-interface
https://w3c.github.io/resource-timing/#extensions-performance-interface
http://webaudio.github.io/web-audio-api/#AudioBufferSourceNode
https://fullscreen.spec.whatwg.org/#api

@wolenetz

**Comment by paulbrucecotton** [comment]
@jdsmith3000 and @wolenetz: How do we triage this issue?  Is it V1 or VNext?

**Comment by wolenetz** [comment]
Nice-to-have, not must-have is my initial opinion, given our triage process and our V1 schedule.
@foolip Can you provide an example where the currently missing MSE event handler attributes could be used for feature detection, where the existence of the MSE object (MediaSource, primarily) isn't sufficient? Such could help make a stronger case for inclusion of this issue into MSE v1.

**Comment by ddorwin** [comment]
I could be wrong, but I thought these were discouraged unless they applied to declarable objects. For example, we removed most such handlers from EME when we switched to objects but left onencrypted because it is useful in the `<video>` tag.

**Comment by foolip** [comment]
@wolenetz The feature detection becomes relevant the first time you add an event that isn't currently supported. The existence of an `onfoo` attribute is normally the only way to detect that the `foo` event is implemented. It's not really relevant for v1, except that people may just assume that they exist.

@ddorwin What are declarable objects? There was a time when `onfoo` attributes were unfashionable, but now the norm seems to be included them. Even something brand new like https://slightlyoff.github.io/ServiceWorker/spec/service_worker/#service-worker-obj has it.

**Comment by ddorwin** [comment]
The relevant EME bug comment: https://www.w3.org/Bugs/Public/show_bug.cgi?id=18928#c1

**Comment by foolip** [comment]
FWIW, I'd make the same argument for EME, and can file an issue if you'd like.

**Comment by ddorwin** [comment]
>@ddorwin What are declarable objects?

I meant objects that can be defined declaratively.

>There was a time when `onfoo` attributes were unfashionable, but now the norm seems to be to include them.

The bug above may have been during that time. Can we get definitive guidance from public-script-coord@ or similar?

>FWIW, I'd make the same argument for EME, and can file an issue if you'd like.

I was just about to say the same thing. If the right thing to do is add them for MSE, we should also add them for EME. It probably makes sense to file an issue and note that it is blocked on resolution of this issue. Thanks.

**Comment by foolip** [comment]
Seems like something that https://w3ctag.github.io/design-principles/ might cover. @domenic @slightlyoff?

**Comment by domenic** [comment]
Yeah, agreed we can try to include it there.

**Comment by wolenetz** [comment]
If I understand correctly, there is no guidance around whether or not these handlers need attributes.
@domenic do you have a rough idea when there might be such guidance (probably in https://github.com/w3ctag/design-principles/issues/29)? Both MSE and EME need it ASAP to be able to effectively triage this issue into V1 or VNext.

**Comment by domenic** [comment]
I gave some guidance in the w3ctag issue you opened.

**Comment by wolenetz** [comment]
@domenic, thanks for the guidance. Based on https://github.com/w3ctag/design-principles/issues/29#issuecomment-215560292, it looks like this is something that needs fixing in MSE v1 scope. @jdsmith3000 / @mwatson2, Do you agree this needs to be marked V1?

**Comment by wolenetz** [comment]
We need to add:
* Attributes for `MediaSource` events:
 * `onsourceopen`
 * `onsourceended`
 * `onsourceclose`
* Attributes for `SourceBuffer` events:
 * `onupdatestart`
 * `onupdate`
 * `onupdateend`
 * `onerror`
 * `onabort`
* Attributes for `SourceBufferList` events:
 * `onaddsourcebuffer`
 * `onremovesourcebuffer`

**Comment by wolenetz** [comment]
FYI - corresponding Chromium change: https://codereview.chromium.org/1945523003/ for Chromium bug https://bugs.chromium.org/p/chromium/issues/detail?id=608912

**Comment by wolenetz** [comment]
https://github.com/wolenetz/media-source/commit/1e4b68c15c327ba96c8f7fffa21b8754bf324499 fixes this. Closing.
